lstchain/calib/camera/flatfield.py

```diff
@@ -278,7 +278,7 @@ def calculate_time_results(
                                              pixel_median > self.time_cut_outliers[1])
 
         return {
-            'sample_time': (trigger_time - time_start).value / 2 *u.s,
+            'sample_time': (trigger_time + time_start).value / 2 *u.s,
```

Comment:
I don't understand what this is supposed to do. Why do you take the mean between two astropy times like this?
Or are the `..._time` here not astropy times? If they are not, what are they?

If you want to have the midpoint, it would correctly be something like this:

```
start_time + 0.5 * (trigger_time - start_time)
```

to get a proper astropy time back and not a value in seconds:

```
In [1]: from astropy.time import Time

In [2]: t0 = Time('2020-01-01T20:00')

In [3]: t1 = Time('2020-01-01T21:00')

In [4]: t0 + 0.5 * (t1 - t0)
Out[4]: <Time object: scale='utc' format='isot' value=2020-01-01T20:30:00.000>
```

Reply:
See also the error in the CI you are getting:

> E           astropy.time.core.OperandTypeError: Unsupported operand type(s) for +: 'Time' and 'Time'

Reply:
The sum of two points in time is not something well-defined. Which is why astropy does not allow it.

The difference is a `TimeDelta` and you can add a `TimeDelta` to a `TimeDelta` or a `TimeDelta` to a `Time`